struqture/src/bosons/bosonic_noise_operator.rs

```diff
@@ -164,6 +171,32 @@ impl<'a> OperateOnDensityMatrix<'a> for BosonLindbladNoiseOperator {
             }
         }
     }
+
+    /// Adds a new entry in the BosonLindbladNoiseOperator with the given ((BosonProduct, BosonProduct) key, CalculatorComplex value) pair.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - The (BosonProduct, BosonProduct) key to set in the BosonLindbladNoiseOperator.
+    /// * `value` - The corresponding CalculatorComplex value to set for the key in the BosonLindbladNoiseOperator.
+    ///
+    /// # Returns
+    ///
+    /// * `Err(StruqtureError)` - The input contained identities, which are not allowed as Lindblad operators.
+    fn add_operator_product(
+        &mut self,
+        key: Self::Index,
+        value: Self::Value,
+    ) -> Result<(), StruqtureError> {
+        if key.0 == BosonProduct::new([], []).unwrap()
+            || key.1 == BosonProduct::new([], []).unwrap()
+        {
+            return Err(StruqtureError::InvalidLindbladTerms);
+        }
+
+        let old = self.get(&key).clone();
+        self.set(key, value + old)?;
+        Ok(())
+    }
```

Comment:
This should automatically use the trait implementation, which internally calls `set` - I think this might just be code duplication